src/main/java/software/amazon/encryption/s3/S3AsyncEncryptionClient.java

```diff
@@ -434,6 +441,22 @@ public Builder enableMultipartPutObject(boolean _enableMultipartPutObject) {
             return this;
         }
 
+        /**
+         * Sets the buffer size for safe authentication used when delayed authentication mode is disabled.
+         * If buffer size is not given during client configuration, default buffer size is set to 64MiB.
+         * @param bufferSize the desired buffer size in Bytes.
+         * @return Returns a reference to this object so that method calls can be chained together.
+         * @throws S3EncryptionClientException if the specified buffer size is outside the allowed bounds
+         */
+        public Builder maxBufferSize(long bufferSize) throws IllegalArgumentException {
```

Comment:
ditto - rename (see other comment)

Reply:
does this ever throw `IllegalArgumentException`?

Reply:
Async class isn't tested; let's at least add some basic tests for the async client.